### DISCUSSION TITLE
which method is stealthiest and most seleniumbase compatible? execute_script(js) or pyautogui or click() without uc?

### DISCUSSION THREAD

**Discussion by bobberystudent** [seleniumbase/SeleniumBase#2787] [Q&A] [answered]
Hello everyone, I found a checkbox in a website yesterday not detectable by seleniumbase that uc_click() method doesnt work on because this checkbox element uses JavaScript for click functionality via `@click `directive in the HTML. after trial an error and additional problemsolving code, i made it work with click() but not uc_click(), and found other ways of workaround via using executescript() to directly execute javascript and click the element, as well as another way via using pyautogui to click element via navigating the website by using the tab key to click. 

id rather do everything in selenium but **which of these 3 methods is most "stealthy" and least likely for detection? js_click()/click(), execute_script(js), or pyautogui?** also is there a way using connect() disconnect() to make js_click() similar to uc_clikc()? I read on previous discussion forms that executescript may raise red flags when doing something like opening a tab, so I am leaning to pyautogui but don't know how seleniumbase compatible it is, or how "stealthy" it is as i lack experience with the library. 

also if there is another better workaround, pls lmk, and thank you everyone

**Comment by mdmintz** [comment, marked as answer]
`driver.uc_click(selector)` is a stealth click in UC Mode.
For maximum stealth, combine UC Mode with `pyautogui`. (Eg: https://github.com/seleniumbase/SeleniumBase/issues/2626#issuecomment-2011102682)

**Reply by bobberystudent** [reply]
is js_click() or pyautgui more stealthy with UC mode?

**Reply by mdmintz** [reply]
`pyautgui` is the stealthiest way.
